src/model/ModelMerger.cpp

```diff
@@ -356,6 +356,29 @@ namespace model {
           }
         }
       }
+
+      // match new to cloned subsurfaces
+      for (const auto& newSubSurface : newSurface.subSurfaces()) {
+        m_newMergedHandles.insert(newSubSurface.handle());
+        for (auto& cloneSubSurface : clone.subSurfaces()) {
+          if (circularEqual(newSubSurface.vertices(), cloneSubSurface.vertices(), 0.01)) {
+            m_currentToNewHandleMapping[cloneSubSurface.handle()] = newSubSurface.handle();
+            m_newToCurrentHandleMapping[newSubSurface.handle()] = cloneSubSurface.handle();
+
+            boost::optional<SubSurface> newAdjacentSubSurface = newSubSurface.adjacentSubSurface();
+            if (newAdjacentSubSurface) {
+              boost::optional<UUID> currentAdjacentSubSurfaceHandle = getCurrentModelHandle(newAdjacentSubSurface->handle());
+              if (currentAdjacentSubSurfaceHandle) {
+                boost::optional<SubSurface> currentAdjacentSubSurface = m_currentModel.getModelObject<SubSurface>(*currentAdjacentSubSurfaceHandle);
+                if (currentAdjacentSubSurface) {
+                  cloneSubSurface.setAdjacentSubSurface(*currentAdjacentSubSurface);
+                }
+              }
+            }
+            break;
+          }
+        }
+      }
```

Comment:
Ok, so:

1. We always emplace subSurfaces in m_newMergedHandles
2. we do resolve m_currentToNewHandleMapping and m_newToCurrentHandleMapping
3. Only if newAdjacentSubSurface then we try to setAdjacentSubSurface

Do we _need_ 1 & 2 if `newAdjacentSubSurface` is empty?

I can see it follows the intent, **but** matching via `circularEqual` is expensive and it could be a big slow down on large models with lots of surfaces & more importantly lots of subsurfaces on each surface.

Reply:
The issue was that `clone` clones all the children without any support for the handle mapping so I tried to find a way to find the same sub surfaces in both models after the clone.  I thought about matching sub surfaces based on names but there are some degenerate cases that would fail on (e.g. merge model {Space1 with SubSurface1, Space2 with SubSurface2} with model {Space1 with SubSurface2, Space2 with SubSurface1} might change object names based on the order spaces are processed).

My thought was to add all sub surfaces to the handle mappings for completeness, but we don't 100% need it.  I would be ok to only do the checks if `adjacentSubSurface` is not empty.  I'll update this tonight.

Changes to clone to support the handle mapping would be possible, and might be generally useful, just a bigger change.

Reply:
I made the change and tested it locally.  I updated my branch and pushed the change however I can no longer build after the branch was updated to latest develop, I don't have my environment set up for 3.8.0 yet.

Reply:
https://github.com/NREL/OpenStudio/blob/develop/BUILDING.md